File: mobile/ios/Accord/Models/VoiceState.swift
// VoiceState.swift — Voice channel participation state

import Foundation

struct VoiceParticipant: Identifiable, Hashable {
    let id: String      // User ID
    var displayName: String
    var isMuted: Bool = false
    var isDeafened: Bool = false
    var isSpeaking: Bool = false
}

@Observable
final class VoiceState {
    var isConnected: Bool = false
    var currentChannelId: String?
    var participants: [VoiceParticipant] = []
    var isSelfMuted: Bool = false
    var isSelfDeafened: Bool = false
}

File: mobile/android/app/src/main/java/com/accord/data/model/VoiceState.kt
package com.accord.data.model

data class VoiceState(
    val channelId: String,
    val participants: List<VoiceParticipant> = emptyList(),
    val isMuted: Boolean = false,
    val isDeafened: Boolean = false,
    val isConnected: Boolean = false,
)

data class VoiceParticipant(
    val userId: String,
    val displayName: String,
    val isMuted: Boolean = false,
    val isSpeaking: Boolean = false,
)
